modules/issue_tracker/jsx/IssueForm.js

```diff
@@ -72,15 +96,39 @@ class IssueForm extends Component {
       dateCreated = 'Sometime Soon!';
       submitButtonValue = 'Submit Issue';
       commentLabel = 'Description';
+      attachmentFileElement = (
+        <FileElement
+          name='file'
+          label='Attachment for issue'
+          onUserInput={this.setFormData}
+          errorMessage={this.state.errorMessage}
+          value={this.state.formData.file}
+        />
+      );
     } else {
       headerText = 'Edit Issue #' + this.state.issueData.issueID;
       lastUpdateValue = this.state.issueData.lastUpdate;
       lastUpdatedByValue = this.state.issueData.lastUpdatedBy;
       dateCreated = this.state.issueData.dateCreated;
       submitButtonValue = 'Update Issue';
       commentLabel = 'New Comment';
+      attachmentUploadBtn = (
+        <ButtonElement
+          onUserInput={this.openAttachmentUploadModal}
+          label={'Add Attachment'}
+        />
+      );
     }
 
+    const fileCollection = this.state.isNewIssue || (
+      <AttachmentsList issue={this.props.issue}
+                       attachments={this.state.issueData['attachments']}
+                       userHasPermission={
+                         loris.userHasPermission('issue_tracker_developer')
```

Comment:
this should be passed as a prop rather than relying on a global

Reply:
@driusan Do you want me to pass the result of loris.userHasPermission('issue_tracker_developer') as a prop from the main component of issue_tracker and to the component AttachmentsList? Instead of passing it to the prop of Component AttachmentsList.

Reply:
It should be passed from wherever the window load handler is that creates the initial component all the way down to anywhere that needs it.

Reply:
Hey @driusan, see latest commit.